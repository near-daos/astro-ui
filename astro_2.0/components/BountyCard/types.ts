import { Token } from 'types/token';
import { BountyStatus } from 'types/bounties';

export enum CardType {
  Bounty = 'Bounty',
  Claim = 'Claim',
}
export interface BountyCardContent {
  id: string;
  daoId: string;
  amount: string;
  type: CardType;
  description: string;
  externalUrl?: string;
  token: Token;
  status: BountyStatus;
  bountyBond: string;
  slots?: number;
  slotsTotal?: number;
  claimedByCurrentUser: boolean;
  accountId?: string;
  timeToComplete: string;
  forgivenessPeriod: string;
  relatedProposalHash?: string;
}
