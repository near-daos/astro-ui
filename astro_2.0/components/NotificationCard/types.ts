import { NotificationType, NotificationStatus } from 'types/notification';

export interface NotificationCardContent {
  id: string;
  type: NotificationType;
  status: NotificationStatus;
  text: string;
  time: string;
  flagCover?: string;
  logo?: string;
  url?: string;
}
