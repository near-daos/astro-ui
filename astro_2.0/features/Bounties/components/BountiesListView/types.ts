import { Bounty } from 'types/bounties';
import { ReactNode } from 'react';
import { ProposalVariant } from 'types/proposal';

export type SectionItem = {
  id: string;
  title: string;
  proposer: string;
  proposalId: string;
  bounty?: Bounty;
  link: {
    pathname: string;
    query: {
      dao: string;
      bounty?: string;
      proposal?: string;
    };
  };
  completeHandler?: (
    id: number,
    variant: ProposalVariant.ProposeDoneBounty
  ) => void;
  content: ReactNode;
};
