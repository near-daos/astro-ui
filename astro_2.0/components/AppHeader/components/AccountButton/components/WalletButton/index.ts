export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletButton/WalletButton';
