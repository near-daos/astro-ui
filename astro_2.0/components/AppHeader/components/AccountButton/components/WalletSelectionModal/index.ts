export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletSelectionModal/WalletSelectionModal';
