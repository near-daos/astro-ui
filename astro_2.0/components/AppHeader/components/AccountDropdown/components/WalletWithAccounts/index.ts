export * from 'astro_2.0/components/AppHeader/components/AccountDropdown/components/WalletWithAccounts/WalletWithAccounts';
