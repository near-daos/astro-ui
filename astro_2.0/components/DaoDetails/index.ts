export * from './DaoDetailsMinimized';
export * from './DaoDetailsGrid';
