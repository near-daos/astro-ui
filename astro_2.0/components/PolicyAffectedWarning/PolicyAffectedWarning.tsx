import isEmpty from 'lodash/isEmpty';
import { useRouter } from 'next/router';
import React, { FC, useCallback } from 'react';
import { useTranslation, Trans } from 'next-i18next';

import { SINGLE_PROPOSAL_PAGE_URL } from 'constants/routing';

import { Proposal, ProposalType } from 'types/proposal';

import { Icon } from 'components/Icon';
import { Button } from 'components/button/Button';

import styles from './PolicyAffectedWarning.module.scss';

export interface PolicyAffectedWarningProps {
  data: Pick<Proposal, 'daoId' | 'id' | 'kind'>[];
  className: string;
}

export const PolicyAffectedWarning: FC<PolicyAffectedWarningProps> = ({
  data,
  className = '',
}) => {
  const router = useRouter();
  const { t } = useTranslation();

  const goToProposalPage = useCallback(() => {
    router.push({
      pathname: SINGLE_PROPOSAL_PAGE_URL,
      query: {
        dao: data[0].daoId,
        proposal: data[0].id,
      },
    });
  }, [data, router]);

  if (isEmpty(data)) {
    return null;
  }

  let title = t('daoConfig');

  if (data[0].kind.type === ProposalType.ChangePolicy) {
    title = t('votingPolicy');
  }

  return (
    <div className={className}>
      <div className={styles.root}>
        <div className={styles.status}>
          <Icon name="info" className={styles.icon} />
        </div>
        <div className={styles.content}>
          <div className={styles.title}>
            <Trans i18nKey="policyAffectedProposalsTitle" values={{ title }} />
          </div>
          <div className={styles.text}>
            <Trans i18nKey="policyAffectedProposalsText" values={{ title }} />
          </div>
        </div>
        <div className={styles.control}>
          {data.length === 1 && (
            <Button variant="primary" onClick={goToProposalPage}>
              {t('viewProposal')}
            </Button>
          )}
        </div>
      </div>
    </div>
  );
};
