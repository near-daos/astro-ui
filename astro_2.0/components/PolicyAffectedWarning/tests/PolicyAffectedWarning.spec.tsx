/* eslint-disable @typescript-eslint/ban-ts-comment */

import { render } from 'jest/testUtils';
import { useRouter } from 'next/router';
import { fireEvent } from '@testing-library/dom';

import { Proposal, ProposalType } from 'types/proposal';

import { SINGLE_PROPOSAL_PAGE_URL } from 'constants/routing';

import { PolicyAffectedWarning } from 'astro_2.0/components/PolicyAffectedWarning';

jest.mock('next/router', () => {
  return {
    useRouter: jest.fn(),
  };
});

jest.mock('next-i18next', () => ({
  Trans: ({
    i18nKey,
    values,
  }: {
    i18nKey: string;
    values: Record<string, unknown>;
  }) => <div>{`${i18nKey} ${Object.values(values).join(',')}`}</div>,
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string): string => str,
    };
  },
}));

describe('policy affected warning', () => {
  function getProposal(props?: Record<string, unknown>) {
    return {
      daoId: '123',
      id: '123',
      ...props,
    } as Proposal;
  }

  it('Should render nothing if data is empty', () => {
    const { container } = render(<PolicyAffectedWarning data={[]} />);

    expect(container).toMatchSnapshot();
  });

  it('Should render component', () => {
    const { container, getAllByText } = render(
      <PolicyAffectedWarning data={[getProposal()]} />
    );

    expect(container).toMatchSnapshot();
    expect(getAllByText('daoConfig', { exact: false })).toHaveLength(2);
  });

  it('Should render "votingPolicy" title', () => {
    const { getAllByText } = render(
      <PolicyAffectedWarning
        data={[
          getProposal({
            kind: {
              type: ProposalType.ChangePolicy,
            },
          }),
        ]}
      />
    );

    expect(getAllByText('votingPolicy', { exact: false })).toHaveLength(2);
  });

  it('Should navigate to proposal page', () => {
    const router = {
      push: jest.fn(),
    };

    // @ts-ignore
    useRouter.mockImplementation(() => router);

    const { getByRole } = render(
      <PolicyAffectedWarning data={[getProposal()]} />
    );

    const button = getByRole('button');

    fireEvent.click(button);

    expect(router.push).toBeCalledWith({
      pathname: SINGLE_PROPOSAL_PAGE_URL,
      query: { dao: '123', proposal: '123' },
    });
  });
});

/* eslint-enable @typescript-eslint/ban-ts-comment */
