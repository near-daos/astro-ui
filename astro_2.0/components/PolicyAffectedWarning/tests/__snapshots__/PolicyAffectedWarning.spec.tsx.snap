// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`policy affected warning Should render component 1`] = `
<div>
  <div>
    <div
      class="root"
    >
      <div
        class="status"
      >
        <svg
          class="icon"
        >
          <use
            href="/_next/undefined"
          />
        </svg>
      </div>
      <div
        class="content"
      >
        <div
          class="title"
        >
          <div>
            policyAffectedProposalsTitle daoConfig
          </div>
        </div>
        <div
          class="text"
        >
          <div>
            policyAffectedProposalsText daoConfig
          </div>
        </div>
      </div>
      <div
        class="control"
      >
        <button
          class="btn primary sizeMedium capitalize"
          type="button"
        >
          viewProposal
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`policy affected warning Should render nothing if data is empty 1`] = `<div />`;
